File: 백준/Silver/10866. 덱/덱.swift
let n = Int(readLine()!)!
var deque: [Int] = []

func push_front(_ x: Int) {
    deque.insert(x, at: 0)
}
func push_back(_ x: Int) {
    deque.append(x)
}
func pop_front() {
    if deque.isEmpty {
        print("-1")
    } else {
        print(deque.remove(at: 0))
    }
}
func pop_back() {
    if deque.isEmpty {
        print("-1")
    } else {
        print(deque.popLast()!)
    }
}
func size() {
    print(deque.count)
}
func empty() {
    if deque.isEmpty {
        print("1")
    } else {
        print("0")
    }
}
func front() {
    if deque.isEmpty {
        print("-1")
    } else {
        print(deque.first!)
    }
}
func back() {
    if deque.isEmpty {
        print("-1")
    } else {
        print(deque.last!)
    }
}


for _ in 0..<n {
    let cmd = readLine()!.split(separator: " ")
    
    switch cmd[0] {
    case "push_front":
        if let x = Int(cmd[1]) {
            push_front(x)
        }
    case "push_back":
        if let x = Int(cmd[1]) {
            push_back(x)
        }
    case "pop_front":
        pop_front()
    case "pop_back":
        pop_back()
    case "size":
        size()
    case "empty":
        empty()
    case "front":
        front()
    case "back":
        back()
    default:
        break
    }
}

File: 백준/Silver/10866. 덱/덱.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.List;
import java.util.Deque;

public class Main {
    public static Deque<Integer> listDeque = new LinkedList<>();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        while (N-- > 0) {
            String[] command = br.readLine().split(" ");

            switch (command[0]) {
                case "push_front":
                    push_front(Integer.parseInt(command[1]));
                    break;
                case "push_back":
                    push_back(Integer.parseInt(command[1]));
                    break;
                case "pop_front":
                    pop_front();
                    break;
                case "pop_back":
                    pop_back();
                    break;
                case "size":
                    size();
                    break;
                case "empty":
                    empty();
                    break;
                case "front":
                    front();
                    break;
                case "back":
                    back();
                    break;
            }
        }
    }

    public static void push_front(int x) {
        listDeque.addFirst(x);
    }

    public static void push_back(int x) {
        listDeque.addLast(x);
    }

    public static void pop_front() {
        if (listDeque.isEmpty()) {
            System.out.println(-1);
        } else {
            System.out.println(listDeque.removeFirst());
        }
    }

    public static void pop_back() {
        if (listDeque.isEmpty()) {
            System.out.println(-1);
        } else {
            System.out.println(listDeque.removeLast());
        }
    }

    public static void size() {
        System.out.println(listDeque.size());
    }

    public static void empty() {
        System.out.println(listDeque.isEmpty() ? 1 : 0);
    }

    public static void front() {
        if (listDeque.isEmpty()) {
            System.out.println(-1);
        } else {
            System.out.println(listDeque.getFirst());
        }
    }

    public static void back() {
        if (listDeque.isEmpty()) {
            System.out.println(-1);
        } else {
            System.out.println(listDeque.getLast());
        }
    }
}